app/screens/home/search/results/results.tsx

```diff
@@ -186,9 +198,18 @@ const SearchResults = ({
         const container: StyleProp<ViewStyle> = fileInfos.length > 1 ? styles.container : undefined;
         const isSingleImage = orderedFilesForGallery.length === 1 && (isImage(orderedFilesForGallery[0]) || isVideo(orderedFilesForGallery[0]));
         const isReplyPost = false;
+        const optionSelected = (selectedItemNumber !== undefined) && (selectedItemNumber === filesForGalleryIndexes[item.id!]);
+        const onLayout = (event: LayoutChangeEvent) => {
+            if (selectedItemNumber === filesForGalleryIndexes[item.id!]) {
+                const {height} = dimensions;
+                setOpenUp(event.nativeEvent.layout.y > height / 2);
```

Comment:
This is not accurate at all, uou should measure where the item is in the screen the moment you tap to open the menu

Reply:
You're correct.  That's a miscalculation on my part.  What this does is calculate the y location, relative to the flatlist which is not what is needed. What I need to get is the the y location relative to the screen. Would that be through a ref or some wrapper component?

Reply:
OnLayout won't recalculate when the component is scrolled that is the main reason for this being inaccurate see https://reactnative.dev/docs/direct-manipulation#measurecallback that said, if the view is animated, we do use this (take it as a reference do not modify) https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/utils/gallery/index.ts#L104 tske into account that if the view is not on screen it will throw, thus the try/catch.. the shared values here you could use to determine position and also use them in animations etc..

Reply:
Thanks, @enahum. That's very helpful.